src/utils.ts

```diff
@@ -1,7 +1,30 @@
-// Util to normalize windows paths to posix
-export function normalizeWindowsPath (input: string = '') {
-  if (!input.includes('\\')) {
-    return input
+import { basename } from 'pathe'
+
+export function resolveAliases (_aliases: Record<string, string>) {
+  // Sort aliases from specific to general (ie. fs/promises before fs)
+  const aliases = Object.fromEntries(Object.entries(_aliases).sort(([a], [b]) =>
+    (b.split('/').length - a.split('/').length) || (b.length - a.length)
+  ))
+  // Resolve alias values in relation to each other
+  for (const key in aliases) {
+    for (const alias in aliases) {
+      if (!['~', '@', '#'].includes(alias[0])) { continue }
+      if (alias === '@' && !aliases[key].startsWith('@/')) { continue } // Don't resolve @foo/bar
+
+      if (aliases[key].startsWith(alias)) {
+        aliases[key] = aliases[alias] + aliases[key].slice(alias.length)
+      }
+    }
   }
-  return input.replace(/\\/g, '/')
+  return aliases
+}
+
+export function sortPaths (paths: string[]) {
+  return paths.sort((a, b) =>
+    (b.split('/').length - a.split('/').length) || (b.length - a.length)
+  )
+}
+
+export function filename (path: string) {
+  return basename(path).replace(/\.[^.]+$/, '')
```

Comment:
I'm not sure of any possible edge-cases but we might as well optimize this with one regex  to extract file name from path.